picard/util/lrucache.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+# Copyright (C) 2017 Antonio Larrosa <alarrosa@suse.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+"""
+Helper class to cache items using a Least Recently Used policy.
+
+It's originally used to cache generated pixmaps in the CoverArtBox object
+but it's generic enough to be used for other purposes if necessary.
+The cache will never hold more than max_size items and the item least
+recently used will be discarded.
+
+>>> cache = LRUCache(3)
+>>> cache['item1'] = 'some value'
+>>> cache['item2'] = 'some other value'
+>>> cache['item3'] = 'yet another value'
+>>> cache['item1']
+'some value'
+>>> cache['item4'] = 'This will push item 2 out of the cache'
+>>> cache['item2']
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+  File "lrucache.py", line 48, in __getitem__
+    return super(LRUCache, self).__getitem__(key)
+KeyError: 'item2'
+>>> cache['item5'] = 'This will push item3 out of the cache'
+>>> cache['item3']
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+  File "lrucache.py", line 48, in __getitem__
+    return super(LRUCache, self).__getitem__(key)
+KeyError: 'item3'
+>>> cache['item1']
+'some value'
+"""
+
+
+class LRUCache(dict):
```

Comment:
PyQt provides a [QPixmapCache](http://pyqt.sourceforge.net/Docs/PyQt4/qpixmapcache.html) class, is that something we could use instead of rolling our own? (If we need an lru cache in the future, we can use the one in functools if that future involves python 3).